src/Content/Translation.php

```diff
@@ -0,0 +1,176 @@
+	 * @codeCoverageIgnore
+	 */
+	public function __debugInfo(): array
+	{
+		return $this->toArray();
+	}
+
+	/**
+	 * Returns the language code of the
+	 * translation
+	 *
+	 * @deprecated 5.0.0 Use `::language()->code()` instead
+	 */
+	public function code(): string
+	{
+		Helpers::deprecated('`$translation->code()` has been deprecated. Use `$translation->language()->code()` instead.', 'translation-methods');
+		return $this->language->code();
+	}
+
+	/**
+	 * Returns the translation content
+	 * as plain array
+	 *
+	 * @deprecated 5.0.0 Use `::version()->content()->toArray()` instead
+	 */
+	public function content(): array
+	{
+		Helpers::deprecated('`$translation->content()` has been deprecated. Use `$translation->version()->content()` instead.', 'translation-methods');
+		return $this->version->content($this->language)->toArray();
+	}
+
+	/**
+	 * Absolute path to the translation content file
+	 *
+	 * @deprecated 5.0.0
+	 */
+	public function contentFile(): string
+	{
+		Helpers::deprecated('`$translation->contentFile()` has been deprecated. Use `$translation->version()->contentFile()` instead.', 'translation-methods');
+		return $this->version->contentFile($this->language);
+	}
+
+	/**
+	 * Creates a new Translation for the given model
+	 */
+	public static function create(
+		ModelWithContent $model,
+		Version $version,
+		Language $language,
+		array $fields,
+		string|null $slug = null
+	): static {
+		// add the custom slug to the fields array
+		if ($slug !== null) {
+			$fields['slug'] = $slug;
+		}
+
+		$version->create($fields, $language);
+
+		return new static(
+			model: $model,
+			version: $version,
+			language: $language,
+		);
+	}
+
+	/**
+	 * Checks if the translation file exists
+	 *
+	 * @deprecated 5.0.0 Use `::version()->exists()` instead
+	 */
+	public function exists(): bool
+	{
+		Helpers::deprecated('`$translation->exists()` has been deprecated. Use `$translation->version()->exists()` instead.', 'translation-methods');
+		return $this->version->exists($this->language);
+	}
+
+	/**
+	 * Returns the translation code as id
+	 */
+	public function id(): string
+	{
+		return $this->language->code();
+	}
+
+	/**
+	 * Checks if the this is the default translation
+	 * of the model
+	 *
+	 * @deprecated 5.0.0 Use `::language()->isDefault()` instead
+	 */
+	public function isDefault(): bool
+	{
+		Helpers::deprecated('`$translation->isDefault()` has been deprecated. Use `$translation->language()->isDefault()` instead.', 'translation-methods');
+		return $this->language->isDefault();
+	}
+
+	/**
+	 * Returns the language
+	 */
+	public function language(): Language
+	{
+		return $this->language;
+	}
+
+	/**
+	 * Returns the parent page, file or site object
+	 */
+	public function model(): ModelWithContent
+	{
+		return $this->model;
+	}
+
+	/**
+	 * Returns the custom translation slug
+	 */
+	public function slug(): string|null
+	{
+		return $this->version->read($this->language)['slug'] ?? null;
```

Comment:
`Suggestion`: As we want to cache the `Content` object in `Version`, we can increase performance by using it here:

```suggestion
		return $this->version->content($this->language)->data()['slug'] ?? null;
```